megamek/src/megamek/common/weapons/MissileWeaponHandler.java

```diff
@@ -400,10 +399,9 @@ protected int calcAttackValue() {
                 
         //Point Defenses engage the missiles still aimed at us
         if (ae.usesWeaponBays() || ae.isCapitalFighter()) {
-            // Grounded dropships should fire with individual AMS, not bays
-            counterAV = calcCounterAV();
+            CounterAV = calcCounterAV();
         }
-        av = av - counterAV;
+        av = av - CounterAV;
```

Comment:
It wasn't just about adding a local variable. It was that you had three lines of code where only one is needed. Now it's two lines. If you're only changing the value of av if it meets the previous condition, then put the arithmetic operation inside the if block. As written this is effectively identical to
```
if (ae.usesWeaponBays() || ae.isCapitalFighter()) {
    av = av - calcCounterAV();
} else {
    av = av - 0;
}
```
Instead just use `av = av - calcCounterAV();` as the body of the if block.

Reply:
Let me look at it some more.  Part of the reason this is written as it is is that I don't want calcCounterAV() being called unless we meet the usesWeaponBays or isCapitalFighter conditions.  Though a grounded dropship does not *use* weapon bays it has them (they have to exist so that they can be enabled at liftoff) , and that will make calcCounterAV run successfully when it shouldn't.     Part of my goal is to allow this code to work for mobile structures and large naval support vees later.

Reply:
Ok, I think this is fixed now, with a couple of other errors affecting fighter squadrons with aero sanity that I found in testing.